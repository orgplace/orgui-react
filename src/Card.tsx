import classNames from 'classnames';
import React from 'react';
import { OuiColorVariationAttributes } from '.';

type OuiCardAttributes = React.HTMLAttributes<HTMLElement> & OuiColorVariationAttributes

const OuiCard: React.FC<OuiCardAttributes> = (props) => {
  const {colorVariation, ...divProps} = props;
  return (
    <div {...divProps} className={classNames(
      'ouiCard',
      colorVariation,
      divProps.className,
    )}>{
      React.Children.map(divProps.children, (child) => {
        const childElm = child as React.ReactElement;
        return React.cloneElement(childElm, {
          className: classNames(childElm.props.className, 'ouiCardItem'),
        })
      })
    }</div>
  );
}

const OuiShorthandCard: React.FC<OuiCardAttributes> = (props) => {
  const {colorVariation, ...divProps} = props;
  return (
    <div {...divProps} className={classNames(
      'ouiCard ouiCardItem',
      colorVariation,
      divProps.className,
    )}>{divProps.children}</div>
  );
}

export { OuiCard, OuiShorthandCard };
