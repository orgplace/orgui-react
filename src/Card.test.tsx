import React from 'react';
import renderer from 'react-test-renderer';
import { OuiCard } from './Card';

test('OuiCard', function() {
	const component = renderer.create(
		<OuiCard>
			<h1>Sample</h1>
			<div>Contents</div>
		</OuiCard>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
