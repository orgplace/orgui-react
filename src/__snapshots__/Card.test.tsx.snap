// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiCard 1`] = `
<div
  className="ouiCard"
>
  <h1
    className="ouiCardItem"
  >
    Sample
  </h1>
  <div
    className="ouiCardItem"
  >
    Contents
  </div>
</div>
`;
